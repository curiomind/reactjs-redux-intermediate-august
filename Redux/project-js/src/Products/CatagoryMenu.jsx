import React from 'react';
import PropTypes from 'prop-types';
import { Nav } from 'react-bootstrap';

import { Catagories } from '../shared/constants/Catagories';

import CatagoryItem from './CatagoryItem';

function CatagoryMenu({ selected, onCatagoryClick }) {
  return (
    <Nav variant="pills" defaultActiveKey="/home" className="mb-4 pe-4">
      {Catagories.map((catagory) => (
        <CatagoryItem key={catagory.id} catagory={catagory} selected={selected} onCatagoryClick={onCatagoryClick} />
      ))}
    </Nav>
  );
}

CatagoryItem.propTypes = {
  selected: PropTypes.string.isRequired,
  onCatagoryClick: PropTypes.func.isRequired,
};

CatagoryMenu.whyDidYouRender = {
  logOnDifferentValues: true,
  customName: 'WDYR-CatagoryMenu',
};

export default React.memo(CatagoryMenu);
