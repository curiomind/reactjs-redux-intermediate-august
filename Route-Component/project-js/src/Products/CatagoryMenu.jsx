import React from 'react';
import PropTypes from 'prop-types';
import { Nav } from 'react-bootstrap';

import { Catagories } from '../shared/constants/Catagories';

import CatagoryItem from './CatagoryItem';

function CatagoryMenu({ selected, onCatagoryClick }) {
  return (
    <Nav variant="pills" defaultActiveKey="/home" className="mb-4">
      {Catagories.map((catagory, index) => (
        <CatagoryItem key={catagory.id} catagory={catagory} selected={selected} onCatagoryClick={onCatagoryClick} />
      ))}
    </Nav>
  );
}

CatagoryItem.propTypes = {
  selected: PropTypes.string.isRequired,
  onCatagoryClick: PropTypes.func.isRequired,
};

export default React.memo(CatagoryMenu);
