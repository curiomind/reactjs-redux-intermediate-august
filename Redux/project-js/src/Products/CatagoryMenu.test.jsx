import React from 'react';
import renderer from 'react-test-renderer';
import CatagoryMenu from './CatagoryMenu';

const selected = 'mobile';
const onCatagoryClick = jest.fn();

describe('Catagory Menu', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<CatagoryMenu selected={selected} onCatagoryClick={onCatagoryClick} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
