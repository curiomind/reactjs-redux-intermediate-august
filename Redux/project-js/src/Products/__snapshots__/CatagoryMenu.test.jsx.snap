// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Catagory Menu renders correctly 1`] = `
<div
  className="mb-4 pe-4 nav nav-pills"
  onKeyDown={[Function]}
>
  <div
    className="nav-item"
  >
    <button
      className="nav-link"
      data-rb-event-key={null}
      disabled={false}
      onClick={[Function]}
    >
      Grocery
    </button>
  </div>
  <div
    className="nav-item"
  >
    <button
      className="nav-link"
      data-rb-event-key={null}
      disabled={false}
      onClick={[Function]}
    >
      Mobiles
    </button>
  </div>
  <div
    className="nav-item"
  >
    <button
      className="nav-link"
      data-rb-event-key={null}
      disabled={false}
      onClick={[Function]}
    >
      Fashion
    </button>
  </div>
  <div
    className="nav-item"
  >
    <button
      className="nav-link"
      data-rb-event-key={null}
      disabled={false}
      onClick={[Function]}
    >
      Electronics
    </button>
  </div>
  <div
    className="nav-item"
  >
    <button
      className="nav-link"
      data-rb-event-key={null}
      disabled={false}
      onClick={[Function]}
    >
      Appliances
    </button>
  </div>
</div>
`;
